ra/ra.go

```diff
@@ -166,6 +167,12 @@ func NewRegistrationAuthorityImpl(
 	}, []string{"limit", "decision"})
 	stats.MustRegister(rlCheckLatency)
 
+	overrideUsageGauge := prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "ratelimitsv1_override_usage",
+		Help: "Proportion of override limit used, by limit name and client identifier.",
+	}, []string{"limit", "override_key"})
```

Comment:
The `override_key` in two particular instances for production `certificatesPerFQDNSet` will be 377 bytes. By default prometheus [doesn't limit the number of labels, label size, etc](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config). We don't set these values in either staging or prod...yet.
```
# Per-scrape limit on length of labels name that will be accepted for a sample.
  # If a label name is longer than this number post metric-relabeling, the entire
  # scrape will be treated as failed. 0 means no limit.
  [ label_name_length_limit: [<int>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#int) | default = 0 ]

  # Per-scrape limit on length of labels value that will be accepted for a sample.
  # If a label value is longer than this number post metric-relabeling, the
  # entire scrape will be treated as failed. 0 means no limit.
  [ label_value_length_limit: [<int>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#int) | default = 0 ]
```

Tangentially related and mostly for posterity, Cloudflare has an interesting [blog post](https://blog.cloudflare.com/how-cloudflare-runs-prometheus-at-scale/) about running prometheus at scale, specifically the sections "Protecting Prometheus from cardinality explosions" and "Basic limits".

Reply:
We're counting on this not being a cardinality explosion because we do control the full set of labels here: they're the lines of our rate limit overrides config files. That's similar to a (larger) company's list of hostnames, which is a very common metric label. There's no way this would be appropriate if we were logging a metric for every rate limit query, but since we're limiting ourselves to the ones which we have manually approved overrides for, we should be okay.